src/org/jitsi/impl/neomedia/rtp/RTPEncodingImpl.java

```diff
@@ -0,0 +1,469 @@
+        }
+
+
+        boolean requires = false;
+
+        if (!ArrayUtils.isNullOrEmpty(dependencyEncodings))
+        {
+            for (RTPEncodingImpl enc : dependencyEncodings)
+            {
+                if (enc.requires(idx))
+                {
+                    requires = true;
+                    break;
+                }
+            }
+        }
+
+        return requires;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean matches(byte[] buf, int off, int len)
+    {
+        long ssrc = RawPacket.getSSRCAsLong(buf, off, len);
+
+        if (primarySSRC != ssrc && rtxSSRC != ssrc)
+        {
+            return false;
+        }
+
+        if (temporalId == -1)
+        {
+            return true;
+        }
+
+        int tid = track.getMediaStreamTrackReceiver()
+            .getStream().getTemporalID(buf, off, len);
+
+        return tid == temporalId;
+    }
+
+    /**
+     * Gets a boolean flag that indicates whether or not this instance is
+     * streaming or if it's suspended.
+     *
+     * @param active true if this {@link RTPEncodingImpl} is active, otherwise
+     * false.
+     */
+    void setActive(boolean active)
+    {
+        this.active = active;
+    }
+
+    /**
+     *
+     * @param pkt
+     * @param nowMs
+     *
+     * @return the {@link FrameDesc} that was updated, otherwise null.
+     */
+    FrameDesc update(RawPacket pkt, long nowMs)
+    {
+        // Update rate stats.
+        rateStatistics.update(pkt.getLength(), nowMs);
+
+        long ts = pkt.getTimestamp();
+        FrameDesc frame = frames.get(ts);
+
+        if (frame == null)
+        {
+            synchronized (frames)
+            {
+                frames.put(ts, frame = new FrameDesc(this, ts));
+            }
+
+            // We measure the stable bitrate on every new frame.
+            lastStableBitrateBps = getBitrateBps(nowMs);
+        }
+
+        // Update the frame description.
+        boolean frameChanged = frame.update(pkt);
+        if (frameChanged)
+        {
+            // Frame boundaries heuristics.
+
+            // Find the closest next frame.
+            Map.Entry<Long, FrameDesc> ceilingEntry
+                = frames.ceilingEntry((ts + 1) & 0xFFFFFFFFL);
+
+            if (ceilingEntry != null)
+            {
+                FrameDesc ceilingFrame = ceilingEntry.getValue();
+
+                // Make sure the frames are close enough, both in time and in
+                // sequence numbers.
+                long tsDiff = (ceilingFrame.getTimestamp() - ts) & 0xFFFFFFFFL;
+
+                if (tsDiff < (1L<<31)
+                    && (ceilingFrame.getStart() != -1 || frame.getEnd() != -1))
+                {
+                    int minSeen = ceilingFrame.getMinSeen();
+                    int maxSeen = frame.getMaxSeen();
+                    int snDiff = (minSeen - maxSeen) & 0xFFFF;
+
+                    // We can deal with distances that are less than 3.
```

Comment:
> The problem with this heuristic is that we may end up not accounting for a frame of a single packet between the ceiling and the received frame.

I don't see how this is possible if we have not seen the end of 'frame', and we have not seen the start of 'ceilingFrame'. 
minSeen and maxSeen are only updated with the sequence number of an actual receive packet, and are not inferred. So having a value for e.g. minSeen, and having no value for start implies that start < minSeen. Similarly, the the previous frame end > maxSeen. And if diff is 3, then the there are only two available spots for the end of the first frame and the start of the second. No place for a 1-packet frame to hide.